doc/rst/meta/modules/packages.rst

```diff
@@ -6,17 +6,118 @@ Package Modules
 Package modules are libraries that currently live outside of the
 Chapel Standard Library, either because they are not considered to be
 fundamental enough or because they are not yet mature enough for
-inclusion there.
+inclusion there.  Over time, we expect many of these to become
+``mason`` packages.
 
+Algorithms
+----------
+.. toctree::
+   :maxdepth: 1
+
+   Crypto <packages/Crypto>
+   Search <packages/Search>
+   Sort <packages/Sort>
+
+Communication (Inter-Locale)
```

Comment:
I have a slight preference to put everything here under "Parallel Computing". `AtomicObject` is there, but `UnorderedAtomics` here, for example. That feels wrong.

Reply:
Another alternative is to put MPI there, and the rest under something like "Optimizations". But my preference is still everything together under "Parallel Computing"

Reply:
I originally distinguished between "Parallel Computing" and "Distributed Computing" where these four were under the latter.  But then they all seemed to be more specifically about optimizing or controlling communication which is why I regrouped and renamed them as shown currently.  I.e., I don't think you'd use any of these modules for programs that were only intended for a single locale?

AllLocalesBarrier definitely belongs in the same category as these, so that was a mistake on my part due to going too fast.

Is AtomicObject intended primarily for distributed computing, or is it like other Chapel things that support general parallelism and happen to also work in a distributed setting?  (e.g., I'd consider `atomic int` to be more about parallel computing than distributed computing, so put atomic object into the parallel category for similar reasons).

My sense is that Futures aren't particularly geared to distributed execution either, but I could be similarly wrong about that.

The other related tension I had was that I was tempted to sort the data structures by those that are simple vs. those designed for parallel computing vs. those supporting distributed computing, but wasn't certain whether there was a way to create a nested toc so just lumped them all together for now.

Reply:
I should add that I feel more negatively about glomming things relating to parallelism (multiple tasks) into the same category as those related to distributed computing (multiple locales) than I do about the name of this current category.  So if switching this category to a name like "Distributed Computing", "Scalable Parallel Computing", "Scalability", "Inter-locale Coordination and Optimization", "Multi-locale Coordination and Optimization", or the like would help, I'd be up for that.

Another part of my reason for distinguishing them is that I think that the modules in this category are currently a bit more "first-class" / "blessed" central, whereas the ones in the Parallel category feel a bit more half-baked or unexercised (?).  And I liked that the current choice of name put them higher in the document, alphabetizing (but didn't choose it for that reason).

I'd also be curious what @ronawho thinks here since most of these modules are his.

Reply:
I'm on board with the current categories